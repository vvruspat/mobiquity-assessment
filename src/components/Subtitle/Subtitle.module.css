.subtitle {
    padding: 0;
    margin: 0;
    color: var(--regular-color);
}