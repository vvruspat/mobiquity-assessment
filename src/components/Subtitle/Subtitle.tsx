import {
  DetailedHTMLProps,
  FC,
  HTMLAttributes,
  PropsWithChildren,
} from "react";
import cn from "clsx";
import { Text } from "../Text";

import s from "./Subtitle.module.css";

type SubtitleProps = DetailedHTMLProps<
  HTMLAttributes<HTMLHeadingElement>,
  HTMLHeadingElement
>;

/**
 * Subtitle component
 */
export const Subtitle: FC<PropsWithChildren<SubtitleProps>> = ({
  children,
  className,
  ...htmlProps
}) => {
  return (
    <h4 className={cn(s.subtitle, className)} {...htmlProps}>
      <Text size="l" weight="semibold">
        {children}
      </Text>
    </h4>
  );
};
