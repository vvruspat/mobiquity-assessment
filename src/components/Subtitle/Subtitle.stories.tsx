import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Subtitle } from "./Subtitle";
import "../../index.css";

export default {
  title: "UI Kit/Typography/Subtitle",
  component: Subtitle,
} as ComponentMeta<typeof Subtitle>;

const Template: ComponentStory<typeof Subtitle> = (args) => (
  <Subtitle {...args}>Subtitle</Subtitle>
);

export const SubtitleTemplate = Template.bind({});

SubtitleTemplate.args = {};
