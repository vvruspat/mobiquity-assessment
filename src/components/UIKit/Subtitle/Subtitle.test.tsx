import { render, screen } from "@testing-library/react";
import { Subtitle } from "./Subtitle";

test("renders subtitle", () => {
  render(<Subtitle>Test subtitle</Subtitle>);

  const el = screen.getByTestId("subtitle");

  expect(el).toHaveClass("subtitle");
  expect(el).toHaveTextContent("Test subtitle");
  expect(el).toMatchSnapshot();
});
