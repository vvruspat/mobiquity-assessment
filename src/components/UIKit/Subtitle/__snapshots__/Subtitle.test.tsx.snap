// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders subtitle 1`] = `
<h4
  class="subtitle"
  data-testid="subtitle"
>
  <span
    class="text size_l semibold"
    data-testid="text"
  >
    Test subtitle
  </span>
</h4>
`;
